libs/remix-core-plugin/src/lib/compiler-content-imports.ts

```diff
@@ -16,12 +16,39 @@ export type ResolvedImport = {
 }
 
 export class CompilerImports extends Plugin {
-  previouslyHandled: Record<string, ResolvedImport>
   urlResolver: any
   constructor () {
     super(profile)
-    this.urlResolver = new RemixURLResolver()
-    this.previouslyHandled = {} // cache import so we don't make the request at each compilation.
+    this.urlResolver = new RemixURLResolver(async () => {
+      try {
+        let yarnLock
+        if (await this.call('fileManager', 'exists', './yarn.lock')) {
+          yarnLock = await this.call('fileManager', 'readFile', './yarn.lock')
+        }
+
+        let packageLock
+        if (await this.call('fileManager', 'exists', './package-lock.json')) {
+          packageLock = await this.call('fileManager', 'readFile', './package-lock.json')
+          packageLock = JSON.parse(packageLock)
```

Comment:
Needs to be tested with a large `package-lock.json` file

Reply:
Very true. We need an example.